src/Symfony/Component/Console/Command/Command.php

```diff
@@ -250,12 +253,22 @@ public function run(InputInterface $input, OutputInterface $output)
 
         $input->validate();
 
-        if ($this->code) {
-            $statusCode = call_user_func($this->code, $input, $output);
-        } else {
-            $statusCode = $this->execute($input, $output);
+        try {
+            if (!$this->lock($output)) {
+                $statusCode = 0;
+            } elseif ($this->code) {
+                $statusCode = call_user_func($this->code, $input, $output);
+            } else {
+                $statusCode = $this->execute($input, $output);
+            }
+        } catch (\Exception $e) {
+            $this->release();
+
+            throw $e;
         }
 
+        $this->release();
```

Comment:
We also can't use `finally` when autoloading may be triggered, because php has a bug on 5.5 and we have to live with it since it's our lowest php supported version (I don't know if it applies here).